github-connector/internal/registration/registerservice.go

```diff
@@ -1,1020 +1,123 @@
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/issues/2"
-											}
-										}
-									},
-									"comments": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/issues/2/comments"
-											}
-										}
-									},
-									"review_comments": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/pulls/2/comments"
-											}
-										}
-									},
-									"review_comment": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/pulls/comments{/number}"
-											}
-										}
-									},
-									"commits": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/pulls/2/commits"
-											}
-										}
-									},
-									"statuses": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/statuses/ec26c3e57ca3a959ca5aad62de7213c562f8c821"
-											}
-										}
-									}
-								}
-							},
-							"author_association": {
-								"type": "string",
-								"example": "OWNER"
-							}
-						}
-					},
-					"head": {
-						"type": "object",
-						"properties": {
-							"label": {
-								"type": "string",
-								"example": "Codertocat:changes"
-							},
-							"ref": {
-								"type": "string",
-								"example": "changes"
-							},
-							"sha": {
-								"type": "string",
-								"example": "ec26c3e57ca3a959ca5aad62de7213c562f8c821"
-							},
-							"user": {
-								"$ref": "#/components/schemas/user"
-							},
-							"repo": {
-								"$ref": "#/components/schemas/repository"
-							}
-						}
-					},
-					"base": {
-						"type": "object",
-						"properties": {
-							"label": {
-								"type": "string",
-								"example": "Codertocat:master"
-							},
-							"ref": {
-								"type": "string",
-								"example": "master"
-							},
-							"sha": {
-								"type": "string",
-								"example": "f95f852bd8fca8fcc58a9a2d6c842781e32a215e"
-							},
-							"user": {
-								"$ref": "#/components/schemas/user"
-							},
-							"repo": {
-								"$ref": "#/components/schemas/repository"
-							}
-						}
-					}
-				}
-			}
-		}`),
-	},
+//Builder is an interface containing all necessary functions required to build an ServiceDetails structure
+type Builder interface {
```

Comment:
I think the interface should be simplified. I would remove GetApplicationRegistryURL and pass the url in arguments to  NewServiceRegister (NewApplicationRegistryClient). 
I would rename the interface to PayloadBuilder and BuildServiceDetails method to Build.